agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,246 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2020 MaxLinear
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "ieee802_3_link_metrics_collector.h"
+
+#include <bcl/network/network_utils.h>
+
+#include <easylogging++.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <net/if.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+// SPEED values
+#include <linux/ethtool.h>
+
+#define IFLIST_REPLY_BUFFER 8192
+
+namespace beerocks {
+
+/**
+ * @brief Netlink request type.
+ */
+struct nl_req_t {
+    /**
+     * Netlink message
+     */
+    struct nlmsghdr hdr;
+
+    /**
+     * "general form of address family dependent" message, i.e. how to tell which AF we are
+     * interested in. */
+    struct rtgenmsg gen;
+};
+
+static bool get_link_metrics(const struct nlmsghdr *h, const std::string &local_interface_name,
```

Comment:
I'm not sure about what you mean by "more specific". If you please suggest a better name I will change it. However, be aware that this code will be refactored and this function will disappear.

Reply:
Can you elaborate on the fact that this will disappear? You mean it will be part of the discovery service? In that case it is worth to mention that in the commit message.

Reply:
This function will disappear once we do a refactor of this class to use Netlink classes defined in #900 to get link metrics.